src/tool/xmeta/cpp/models/property_model.h

```diff
@@ -57,9 +65,21 @@ namespace xlang::xmeta
             return m_set_method;
         }
 
+        void set_overridden_property_ref(std::shared_ptr<property_model> const& ref) noexcept;
+
+        compilation_error set_get_method(std::shared_ptr<method_model> const& m);
+
+        compilation_error set_set_method(std::shared_ptr<method_model> const& m);
+
+        void validate(xlang_error_manager & error_manager);
+
+        void resolve(symbol_table & symbols, xlang_error_manager & error_manager, std::string const& qualified_name);
+
     private:
-        property_semantics m_semantic;
-        property_type_semantics m_type;
+        property_modifier m_modifier;
+        model_ref<std::shared_ptr<property_model>> m_implemented_property_ref;
+        type_ref m_type;
+        bool m_is_array;
```

Comment:
I wonder if we should bake array semantics into a "parameter type" model, instead of having to code array semantics into property directly

Reply:
I think we should encode that into the type_ref in model_types. It is the type_refs that can be an array anyway. This should cover this, return types and the parameter types.